questions/models.py

```diff
@@ -294,6 +294,18 @@ class Forecast(models.Model):
         related_name="forecasts",
     )
 
+    class SourceChoices(models.TextChoices):
+        API = "api"
+        UI = "ui"
+
+    # logging the source of the forecast for data purposes
+    source = models.CharField(
+        max_length=30,
+        null=True,
```

Comment:
It's quite annoying to handle `null` vs `empty string` in django - https://docs.djangoproject.com/en/5.1/ref/models/fields/#null

Let's use `null=False, blank=True, default=""` instead

Reply:
Ok, I'm pretty much indifferent. (definitely need blank=True, but indifferent between default being "" vs None)

Reply:
Yeah, they recommend to keep only one option in their official doc:
> Avoid using [null](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.Field.null) on string-based fields such as [CharField](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.CharField) and [TextField](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.TextField). If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string. In most cases, it’s redundant to have two possible values for “no data;” the Django convention is to use the empty string, not NULL. One exception is when a [CharField](https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.CharField) has both unique=True and blank=True set. In this situation, null=True is required to avoid unique constraint violations when saving multiple objects with blank values.

Reply:
The thing pulling towards keeping None as default is that in theory "" could denote something more than None.
Namely, old predictions would default to None - meaning we have No Data At All about where it came from. While "" can denote that we tried to track it and didn't come up with a definitive answer. So "" would only apply to new ones that tried but failed to get tracked, while None implies no tracking was ever even attempted.
Thus default would be None in that case, and "" would mean something difference. But eh, I don't think it matters much and can always be changed later.